docs/docs/software-architecture.md

```diff
@@ -31,139 +31,146 @@ UI - User Interface
+ACCS Class:   
+The ACCS class is the interface between the UI and the HTLC. Values set in the UI are respectively set in the config of the ACCS class needed to perform a swap.
 
-Verification:
-The Verification contains the signature verification sub-component which is consulted to verify the swap partners’ signatures. This sub-component comprises of a sign-transaction interface which locally compares the private key to the public key in the browser. If the verification is successful, the swap is executed. If not, the verification fails, and the refund process is initiated.
+HTLC:   
+The HTLC contains several sub-components. A crucial part here is the secret and hash generator that first generates a random secret that is then cryptographically hashed. In order to further conduct the swap, a hash lock sub-component and a time lock sub-component need to be set up. The hash lock accesses the hash generator to retrieve the hashed password, while the time lock cooperates with the refund process, so as soon as the time lock expires, the refund process can be initialized.
 
+Verification:   
+The Verification contains the signature verification sub-component which is consulted to verify the swap partners’ signatures. This sub-component comprises of a sign-transaction interface which locally compares the private key to the public key in the browser. If the verification is successful, the swap is executed. If not, the verification fails, and the refund process is initiated.
+   
 ### 3.1. UML Diagram
 
-![](./docs/static/img/UML.png)
+![](img/UML_new.svg)   
 Figure 2: UML Diagram (Swapchain, 2020) [4]
 
+   
 ## 4. Use Case
 
 The use case diagram is used to visualize the Swapchain application and its actors.
-
+   
 ### 4.1. Use Case Diagram
 
-![](./docs/static/img/newSW_Architecture.svg)  
+![](img/UseCase.svg)     
 Figure 3: Use Case Diagram (Swapchain, 2020) [4]
-
+   
 ### 4.2. Use Case Description
 
 | Use Case Name           | Scenario                               | Triggering Event                   | Actors            | Related Use Case | Preconditions                                                                                                                                       | Post Conditions                                                     | Flow of Events                                                                                                                                                                                                           | Exception Conditions                                          |
```

Comment:
In Triggering Event: Two HTLCs are created in the respective blockchains which locked by the same secret hash provided by the proposer/proposing party.
In Preconditions:  Users should create an HTLC in their respective blockchains by signing their transactions